Build/IntrospectGolang.pm

```diff
@@ -168,6 +168,8 @@ sub buildinfo  {
   my ($fh) = @_;
   my ($vers, $mod) = rawbuildinfo($fh);
   return undef unless defined $vers;
+  $vers =~ s/^go(.*)/$1/;
+
```

Comment:
So I looked at the current syft code and they do not use the "go" prefix in the version of the synthetic "stdlib" package. So why exactly do we need to change this?

Reply:
The prefix comes from the `rawbuildinfo` subroutine.

I've added a print just before the regex to inspect the outcome of the sub like:

```
# ...
  return undef unless defined $vers;
  use Data::Dumper;
  print(Dumper($vers, $mod));
  $vers =~ s/^go(.*)/$1/;
```

Then running it scanning a SLE-15SP4-Minimal-VM:
```
perl -I. generate_sbom --format cyclonedx --dir /mnt/ 
$VAR1 = 'go1.17.9';
$VAR2 = 'path	github.com/SUSE/connect-ng/suseconnect
mod	github.com/SUSE/connect-ng	(devel)	
';
```

`$vers` includes the go prefix that's not compliant with the version regex in trivy.